### ISSUE TITLE
Hints should zoom alongside page zoom

### ISSUE THREAD

**Issue by ddevault** [qutebrowser/qutebrowser#5203]
I often zoom into a web page because I am further away from the screen than normal and need a little help to read the text. However, I can't really navigate around using hints under these conditions, because the hint letters are not zoomed when the page is zoomed in - and I can't read them either.

**Comment by The-Compiler** [comment]
That sounds reasonable, but I wonder if it also should apply when zoomed out - see #1126.

**Comment by ddevault** [comment]
Perhaps there should be a setting which controlls under what conditions the hints are zoomed with the page - never, zoomed out, zoomed in, and always.

**Comment by user202729** [comment]
Alternatively, if it's hard for you to read everything on the is screen, you should consider use the operating system to zoom in (or change the screen resolution)

**Comment by ddevault** [comment]
>change the screen resolution

That is a terrible solution and you should feel ashamed for suggesting it.

I did increase the font size of my hints, but it's annoying to use when I'm not zoomed in.

**Comment by user202729** [comment]
Why? Are you only *sometimes* far from the screen? / You can change the screen zoom factor instead (on Windows), but if you need magnification then the eyes can't distinguish between the "small" and "very small" pixels. What's the problem?

**Comment by ddevault** [comment]
Yes, I am sometimes far from the screen and sometimes near it. And I am a self-respecting human-ass being, which means I don't use Windows.

**Comment by bitraid** [comment]
Do you use Sway by any change :smile: ?

**Comment by bitraid** [comment]
>I did increase the font size of my hints, but it's annoying to use when I'm not zoomed in.

You've probably already thought of this, but as a half-solution, you could bind some keys to switch between different hint font sizes.

**Comment by bitraid** [comment]
I had the idea of using the new `fonts.default_size` setting, alongside with `zoom.default`, to create a binding that zooms the content and also increases the size of the interface fonts (and one to reset them):
```
config.bind('(', 'set fonts.default_size "14px" ;; set zoom.default "100%"')
config.bind(')', 'set fonts.default_size "24px" ;; set zoom.default "180%"')
```
Ironically, i stepped into #5214, which prevents the change of the hints font size, but i think increasing the size of all the fonts should be even better when being further away from the screen (as long as the same size is used for all fonts).

**Comment by IsaacElenbaas** [comment]
I've just written commands to allow incremental zooming of everything, respective to `zoom.levels`. Thanks for pointing out `zoom.default`, @bitraid.
```python3
from qutebrowser.utils import message
try:
    @cmdutils.register(instance='command-dispatcher', scope='window')
    def zoom_in_all(self):
        lastZoom = int(c.zoom.default[:-1])
        c.zoom.default = c.zoom.levels[min(c.zoom.levels.index(c.zoom.default)+1, len(c.zoom.levels)-1)]
        c.fonts.default_size = str(float(c.fonts.default_size[:-2])/lastZoom*int(c.zoom.default[:-1])) + c.fonts.default_size[-2:]
        message.info(c.zoom.default, replace=True)
    @cmdutils.register(instance='command-dispatcher', scope='window')
    def zoom_out_all(self):
        lastZoom = int(c.zoom.default[:-1])
        c.zoom.default = c.zoom.levels[max(c.zoom.levels.index(c.zoom.default)-1, 0)]
        c.fonts.default_size = str(float(c.fonts.default_size[:-2])/lastZoom*int(c.zoom.default[:-1])) + c.fonts.default_size[-2:]
        message.info(c.zoom.default, replace=True)
except ValueError:pass
```
